### PR TITLE
Feat/4700  tlms for tags

### PR DESCRIPTION
**Pull request by grignards**
See #4700 and #4694.

## Checkpoints

**For developers**

- [x] If relevant, manual tests have been done to ensure the stability of the whole application and that the involved feature works
- [x] The original issue is up to date w.r.t the latest discussions and contains a short summary of the implemented solution
- [x] Automated tests covering the feature or the fix, have been written, deemed irrelevant (give the reason), or an issue has been created to implement the test (give the link)
- [x] Make sure that:
  - if there are modifications in the Tracim configuration files (eg. `development.ini`), they are documented in `backend/doc/setting.md`
  - any migration process required for existing instances is documented
  - relevant people for these changes are notified
- [x] Original authors of the features included in a multi-feature branch (maintenance fixes -> develop, security fixes -> develop, …) should be part of the reviewers, especially if you encountered merge conflicts.

**For code reviewers**

- [x] The code is clear enough
- [x] If there are FIXMEs in the code, related issues are mentioned in the FIXME
- [x] If there are TODOs, NOTEs or HACKs in code, the date and the developer initials are present

**For testers**

- [x] Manual, quality tests have been done

### CODE REVIEW

File: backend/requirements-test.txt [hunk 1]
```diff
@@ -2,7 +2,6 @@
 WebTest==2.0.34
 pytest==5.3.5
 pytest-dotenv==0.4.0
-pytest-pyramid-server==1.7.0
```
Comment (inkhey) [inline backend/requirements-test.txt:5]: Cool :+1:

File: backend/tracim_backend/tests/fixtures.py [hunk 2]
```diff
@@ -97,31 +95,45 @@ def empty_event_queues():
...
+                "-n",
+                "tracim_webserver",
+                "--server-name",
+                "tracim_webserver",
+            ],
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
         )
+        try:
+            process.wait(0.5)
```
Comment (inkhey) [inline backend/tracim_backend/tests/fixtures.py:116]: An explanation of this part may be useful.

File: backend/tracim_backend/lib/crud_hook/hookspec.py [hunk 3]
```diff
@@ -90,3 +97,35 @@ def on_workspace_subscription_deleted(
         self, subscription: WorkspaceSubscription, context: TracimContext
     ) -> None:
         ...
+
+    @hookspec
+    def on_reaction_created(self, reaction: Reaction, context: TracimContext) -> None:
```
Comment (inkhey) [inline backend/tracim_backend/lib/crud_hook/hookspec.py:102]: I don't understand why code about reaction is added in this PR.

**Review by inkhey — COMMENTED** [review]
Just some questions.

File: frontend_lib/src/component/Tags/TagList.jsx [hunk 4]
```diff
@@ -24,26 +29,76 @@ class TagList extends React.Component {
...
+    }
+  }
+
+  updateCheckedState (tag, checked) {
+    this.setState(previousState => {
+      const checkedTagIdList = checked
+        ? [...previousState.checkedTagIdList, tag.tag_id]
+        : previousState.checkedTagIdList.filter(id => id !== tag.tag_id)
+      const tagList = previousState.tagList.sort(this.sortTagList(checkedTagIdList))
+      console.log('FOOO', tag, checkedTagIdList, checked)
```
Comment (gbonaspetti) [inline frontend_lib/src/component/Tags/TagList.jsx:100]: FOOO! :laughing: a forgotten console

**Review by gbonaspetti — CHANGES_REQUESTED** [review]

File: backend/tracim_backend/lib/crud_hook/hookspec.py (hunk 3 above)
Comment (grignards) [reply backend/tracim_backend/lib/crud_hook/hookspec.py:102]: the reaction hooks were added without defining their specs, I saw it while adding mine so I added the reaction ones too.

**Review by inkhey — APPROVED** [review]

**Review by gbonaspetti — APPROVED** [review]
Ok for frontend part

File: frontend_lib/src/component/Tags/TagList.jsx (hunk 4 above)
Comment (grignards) [reply frontend_lib/src/component/Tags/TagList.jsx]: Just removed this line.

**Comment by charline-rageade** [comment]
tested on a branch

File: backend/tracim_backend/lib/core/tag.py [hunk 5]
```diff
@@ -16,6 +16,11 @@
 
 
 class TagLib:
+    """
+    This class allows to manage tags on contents.
```
Comment (raphj) [inline backend/tracim_backend/lib/core/tag.py:20]:
```suggestion
    This class allows managing tags on contents.
```

**Review by raphj — APPROVED** [review]

File: backend/development.ini.sample [hunk 6]
```diff
@@ -462,7 +462,7 @@ website.usage_conditions =
 # To include all notifications, use web.notifications.excluded = ""
 # To exclude all notification of an entity, you can use wildcard syntax: for example "user.*" for user,
 # will exclude those events: user.deleted, user.created, user.modified, user.undeleted...
-; web.notifications.excluded = user.*, workspace.modified, workspace.deleted, workspace.undeleted, workspace_member.modified, content.modified, reaction.*
+; web.notifications.excluded = user.*, workspace.modified, workspace.deleted, workspace.undeleted, workspace_member.modified, content.modified, reaction.*, tag.*
```
Comment (raphj) [inline backend/development.ini.sample:465]: Dans ENTITY_TYPE, add tag and content_tag

**Pull request merged**